src/EdFi.DataManagementService.Api.Tests.Integration/APISchemaFileTests.cs

```diff
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Api.Core.ApiSchema;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+
+namespace EdFi.DataManagementService.Api.Tests.Integration;
+
+[TestFixture]
+public class APISchemaFileTests
+{
+    private JsonNode? _schemaContent;
+    private IApiSchemaProvider _apiSchemaProvider;
+    private StringContent _jsonContent;
+    private Action<IWebHostBuilder> _webHostBuilder;
+
+    [SetUp]
+    public void Setup()
+    {
+        _schemaContent = JsonContentProvider.ReadContent("TestSchemaContent.json");
+        _apiSchemaProvider = A.Fake<IApiSchemaProvider>();
+        A.CallTo(() => _apiSchemaProvider.ApiSchemaRootNode).Returns(_schemaContent!);
+
+        _jsonContent = new(
+            JsonSerializer.Serialize(new { property1 = "property1" }),
+            Encoding.UTF8,
+            "application/json"
+        );
+        _webHostBuilder = (builder) =>
+        {
+            builder.UseEnvironment("Test");
+            builder.ConfigureServices(
+                (collection) =>
+                {
+                    collection.AddTransient((x) => _apiSchemaProvider);
+                }
+            );
+        };
+    }
+
+    [Test]
+    public async Task Should_throw_error_when_no_resourcename_element()
```

Comment:
Is there something special about "no_resourcename"? Why should we throw an error here?

Perhaps better to say that it is _responding_ with an error, since it is not _throwing_ an exception.

Reply:
@stephenfuqua 
We are addressing all errors originating from the "ResourceSchema.cs" file. In the "InvalidResourceSchemas.json" file, we have included resource schemas for various invalid use cases. 
For example, the "no_resourcename" schema does not contain a "resourceName" element. So, 
errors will be thrown from the "ResourceSchema.cs" file when attempting to read the "resourceName" from the schema for a particular resource.

Reply:
This is helpful, so let's just add a comment in the class describing the scenarios so that we don't have to guess at it a year from now or when someone else needs to come in and edit it. Could use a block comment that describes the different scenarios in plain English as you've done above.